crates/ibc/src/core/ics04_channel/handler/timeout.rs

```diff
@@ -105,65 +77,100 @@ pub fn process<Ctx: ChannelReader>(
         });
     }
 
-    let result = if source_channel_end.order_matches(&Order::Ordered) {
-        if packet.sequence < msg.next_sequence_recv {
-            return Err(PacketError::InvalidPacketSequence {
-                given_sequence: packet.sequence,
-                next_sequence: msg.next_sequence_recv,
+    // Verify proofs
+    {
+        let client_id_on_a = conn_end_on_a.client_id().clone();
+        let client_state_of_b_on_a = ctx_a
+            .client_state(&client_id_on_a)
+            .map_err(PacketError::Channel)?;
+
+        // check that timeout height or timeout timestamp has passed on the other end
+        let proof_height = msg.proof_height_on_b;
```

Comment:
Can we remove this variable? I don't think it achieves much, and it doesn't follow the convention.

Reply:
[b9eb033](https://github.com/cosmos/ibc-rs/pull/323/commits/b9eb0335bce88634fb00609d1088ce620791aab6)